File: app/src/main/java/com/example/android/sunshine/app/index/ForecastsFragment.java
package com.example.android.sunshine.app.index;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import com.example.android.sunshine.app.R;

import java.util.ArrayList;
import java.util.Arrays;

import javax.inject.Inject;
import javax.inject.Named;

/**
 * A placeholder fragment containing a simple view.
 */
public class ForecastsFragment extends Fragment {
    @Inject
    IndexView view;

    @Inject
    IndexPresenter presenter;

    @Inject @Named("forecasts")
    RecyclerView forecasts;

    public ForecastsFragment() {
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        setHasOptionsMenu(true);
        return inflater.inflate(R.layout.fragment_main, container, false);
    }

    @Override
    public void onResume() {
        super.onResume();
        presenter.bind(view);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        switch (id) {
            case R.id.do_refresh:
                presenter.didPressRefresh();
                return true;
            case R.id.action_settings:
                presenter.gotForecasts(new ArrayList<IndexContract.Forecast>());
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        presenter.bind(null);
        forecasts = null;
    }
}

File: solshine/src/main/java/ph/codeia/solshine/index/ForecastsFragment.kt
package ph.codeia.solshine.index

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.*
import ph.codeia.solshine.Injector
import ph.codeia.solshine.R
import ph.codeia.solshine.shell.ShellContract
import java.util.*
import javax.inject.Inject

/**
 * This file is a part of the Sunshine-Version-2 project.
 */
class ForecastsFragment : Fragment() {
    @Inject
    internal lateinit var presenter: IndexContract.Synchronization

    @Inject
    internal lateinit var user: IndexContract.Interaction

    @Inject
    internal lateinit var view: IndexContract.Display

    @Inject
    internal lateinit var msg: ShellContract.Feedback

    override fun onCreateView(
            inflater: LayoutInflater?,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? = inflater?.inflate(R.layout.fragment_forecasts, container, false)

    override fun onResume() {
        super.onResume()
        @Suppress("UNCHECKED_CAST")
        (activity as? Injector<Fragment>)?.inject(this)
        setHasOptionsMenu(true)
        presenter.bind(view)
        presenter.fetchForecasts()
    }

    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
        inflater?.inflate(R.menu.index, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean = when (item?.itemId) {
        R.id.do_refresh -> true.apply { user.didPressRefresh() }
        R.id.do_add -> true.apply {
            presenter.forecastsFetched(listOf(Weather("new!", "new!!!", Date(), 10.0, 100.0)))
        }
        else -> super.onOptionsItemSelected(item)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        presenter.bind(null)
    }
}
